gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-healthcheck/src/main/java/io/gravitee/gateway/services/healthcheck/rule/EndpointRuleHandler.java

```diff
@@ -169,37 +170,35 @@ protected RequestOptions prepareHttpClientRequest(URI request, HealthCheckStep s
         return options;
     }
 
-    protected URI createRequest(T endpoint, HealthCheckStep step) {
-        URI targetURI = URI.create(endpoint.getTarget());
-
+    protected URL createRequest(T endpoint, HealthCheckStep step) throws MalformedURLException {
+        URL targetURL = new URL(null, endpoint.getTarget());
```

Comment:
probably better to use the third parameter to define your own url handler.

```java
/**
     * Dummy {@link URLStreamHandler} implementation to avoid unknown protocol issue with default implementation
     * (which knows how to handle only http and https protocol).
     */
    private final URLStreamHandler URL_HANDLER = new URLStreamHandler() {
        @Override
        protected URLConnection openConnection(URL u) {
            return null;
        }
    };
```

Reply:
In our case, we only manage http and https protocol.